reference/hoon-expressions/rune/buc.md

```diff
@@ -7,9 +7,8 @@ aliases = ["docs/reference/hoon-expressions/rune/buc/"]
 The `$` family of runes is used for defining custom types.  Strictly speaking,
 these runes are used to produce 'structures'.  A structure is a compile-time
 value that at runtime can be converted to either an example value (sometimes
-called a 'bunt' value) for its corresponding type, or to a 'factory' (sometimes
-called a 'mold').  An example value is used as a placeholder for sample values,
-among other things.  A factory/mold is used as a data validator.
+called a 'bunt' value) for its corresponding type, or to a 'mold'.  An example value is used as a placeholder for sample values,
+among other things.  A mold is used as a data validator.
```

Comment:
I think this section is a little confusing. A `$spec` is a compile-time representation of a mold. A mold can be "bunted", producing a default value, or used to "clam" some arbitrary noun, validating the shape of that noun (and thereby proving its type to the compiler) or crashing.

Reply:
I am still confused. We have at least five interrelated concepts here: types, structures, `$spec`s, molds, and bunt values. Can you evaluate the correctness of each of the following statements? I'm trying to combine what you wrote with what was already there, so there are some contradictions and this is going to be confusing since I am confused.

1. Types exist on an abstract mathematical level. They are simply sets (in the mathematical sense, not the Hoon sense) of nouns.
2. Structures are built from buc runes at compile time. They give the computer the definition of a type in terms of other basic types built into Hoon or other structures.
3. Alternatively, structures are also just types (and so only exist in a mathematical sense). The distinction arises since Hoon only has a few basic types to start with, and structures are types built from these building blocks.
4. A `$spec` is actually what is meant by #2, which means its either the same thing as a structure, or it is how structures are represented in the computer.
5. Molds are built by the compiler to determine which nouns fit a `$spec` (or structure?) and thus are a member of the type it corresponds to. The runtime can use a mold to either clam a noun or bunt it to produce an example value.
6. When a mold is used to clam a noun, it proves its type to the _runtime_ (you wrote compiler, but I don't understand why a compiler would be feeding nouns to a mold at compile-time).
7. The runtime can be asked to bunt a mold, which produces an example value.

To summarize where on the ladder each of these ideas live:
Mathematical: types, maybe structures?
Compile-time: `$spec`s, molds, maybe structures?
Runtime: clamming and bunting.

Reply:
Any time to answer these questison @joemfb ?

Reply:
Ignore this and check my new commit